fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -329,6 +329,17 @@ private boolean isSurveyCategory(final Integer category) {
         return category.equals(DataTableApiConstant.CATEGORY_PPI);
     }
 
+    private JsonElement addColumn(final String name, final String dataType, final boolean isMandatory, final Integer length) {
+        JsonObject column = new JsonObject();
+        column.addProperty("name", name);
+        column.addProperty("type", dataType);
+        if (dataType.equalsIgnoreCase("string")) {
+            column.addProperty("length", length);
+        }
+        column.addProperty("mandatory", (isMandatory ? "true" : "false"));
```

Comment:
Why the ternary operator here instead of just going with String.valueOf(isMandatory)?

Reply:
Done!